### PR TITLE
Oauth fixes

### PR DESCRIPTION
**Pull request by wilco375**
During the connecting of lux ad mosam to an instance of sofia, I found some bugs in our oauth flow. This PR fixes these. See comments below for more details.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -87,8 +87,17 @@ def archive!
   def self.from_omniauth(auth)
     user = where(provider: auth.provider, uid: auth.uid).first_or_create do |u|
       u.name = auth[:info][:name]
+      u.email = auth[:info][:email]
```
Comment (wilco375) [inline app/models/user.rb:90]: These attributes were only added when synchronizing users, not when logging in via OAuth

File: app/models/user.rb [hunk 2]
```diff
@@ -87,8 +87,17 @@ def archive!
...
+      u.avatar_thumb_url = auth[:info][:avatar_thumb_url]
+      u.birthday = auth[:info][:birthday]
     end
     user.update_role(auth[:info][:groups])
+    user.update(
```
Comment (wilco375) [inline app/models/user.rb:95]: These attributes were only synced when creating the user, not when logging in again

File: db/migrate/20170922175055_add_user_model.rb [hunk 3]
```diff
@@ -1,8 +1,6 @@
 class AddUserModel < ActiveRecord::Migration[5.1]
   def change
     create_table :users do |t|
-      t.string :username, null: false
```
Comment (wilco375) [inline db/migrate/20170922175055_add_user_model.rb:4]: I don't know where this field came from, but it is not in `schema.rb` and not used anywhere in our code.

File: lib/omniauth/strategies/amber_oauth2.rb [hunk 4]
```diff
@@ -21,10 +21,10 @@ def authorize_params
       info do
         {
           uid: raw_info['id'],
-          username: raw_info['attributes']['username'],
```
Comment (wilco375) [inline lib/omniauth/strategies/amber_oauth2.rb:24]: This field is not used

File: lib/omniauth/strategies/amber_oauth2.rb [hunk 5]
```diff
@@ -21,10 +21,10 @@ def authorize_params
...
           name: full_name(raw_info),
-          avatar_url: raw_info['attributes']['avatar_url'],
```
Comment (wilco375) [inline lib/omniauth/strategies/amber_oauth2.rb:26]: This field is not used either

File: lib/omniauth/strategies/amber_oauth2.rb [hunk 6]
```diff
@@ -21,10 +21,10 @@ def authorize_params
...
+          email: raw_info['attributes']['email'],
```
Comment (wilco375) [inline lib/omniauth/strategies/amber_oauth2.rb:25]: This field is used but was previously not added to the user during user creation or updated during login

File: lib/omniauth/strategies/amber_oauth2.rb [hunk 7]
```diff
@@ -21,10 +21,10 @@ def authorize_params
...
           avatar_thumb_url: raw_info['attributes']['avatar_thumb_url'],
+          birthday: raw_info['attributes']['birthday'],
```
Comment (wilco375) [inline lib/omniauth/strategies/amber_oauth2.rb:27]: This field is used but was previously not added to the user during user creation or updated during login

**Review by DrumsnChocolate — APPROVED** [review]

**Pull request merged**